masq/src/commands/financials_command.rs

```diff
@@ -402,319 +377,183 @@ impl FinancialsCommand {
-            distinguished
-        )
-    }
-
-    fn width_precise_calculation(
-        headings: &HeadingsHolder,
-        values_of_accounts: &[Vec<String>],
-    ) -> Vec<usize> {
-        let headings_widths = Self::widths_of_str_values(headings.words.as_slice());
-        let values_widths = Self::figure_out_max_widths(values_of_accounts);
-        Self::yield_bigger_values_from_vecs(headings_widths.len(), headings_widths, values_widths)
-    }
-
-    fn widths_of_str_values<T: AsRef<str>>(headings: &[T]) -> Vec<usize> {
-        headings
-            .iter()
-            .map(|phrase| phrase.as_ref().len())
-            .collect()
-    }
-
-    fn create_subset_of_strings_ignoring_the_ordinal_numbers<S: StringValuesOfAccount>(
-        accounts: &[S],
-        gwei: bool,
-    ) -> Vec<Vec<String>> {
-        accounts
-            .iter()
-            .map(|account| account.string_values(gwei))
-            .collect::<Vec<_>>()
-    }
-
-    fn render_accounts_generic<A: StringValuesOfAccount>(
-        &self,
-        stdout: &mut dyn Write,
-        accounts: Vec<A>,
-        headings: &HeadingsHolder,
-        render_headings_fn: fn(&mut dyn Write, &[(&str, &usize)]),
-        render_account_fn: fn(&mut dyn Write, &[String], &[usize], usize),
-    ) {
-        let preformatted_subset = Self::create_subset_of_strings_ignoring_the_ordinal_numbers(
-            &accounts,
-            headings.is_gwei,
-        );
-        let optimal_widths = Self::width_precise_calculation(headings, &preformatted_subset);
-        render_headings_fn(
-            stdout,
-            &Self::zip_them(headings.words.as_slice(), &optimal_widths),
-        );
-        let mut ordinal_number = 0_usize;
-        preformatted_subset.iter().for_each(|account| {
-            ordinal_number += 1;
-            render_account_fn(stdout, account, &optimal_widths, ordinal_number)
-        });
-    }
-
-    fn write_payable_headings(stdout: &mut dyn Write, h_and_w: &[(&str, &usize)]) {
-        short_writeln!(
-            stdout,
-            "{:<ordinal_num_width$}   {:<wallet_width$}   {:<age_width$}   {:<balance_width$}   {:<hash_width$}",
-            h_and_w[0].0,
-            h_and_w[1].0,
-            h_and_w[2].0,
-            h_and_w[3].0,
-            h_and_w[4].0,
-            ordinal_num_width = h_and_w[0].1,
-            wallet_width = WALLET_ADDRESS_LENGTH,
-            age_width = h_and_w[2].1,
-            balance_width = h_and_w[3].1,
-            hash_width = h_and_w[4].1
+    fn convert_masq_from_gwei_and_dress_well<T>(gwei: T) -> String
+    where
+        T: Display + PartialEq + From<u32> + Div<Output = T> + PartialOrd<T>,
+        <T as Div<T>>::Output: PartialEq<T> + PartialOrd<T> + Display,
+    {
+        let (affected_part, is_positive) = match Self::preformatting_math_analysis(gwei) {
+            Left(result) => return result,
+            Right(tuple) => tuple,
+        };
+        let minimal_padding = if is_positive { 3 } else { 4 };
```

Comment:
Two things here: first, it looks like this line is designed to leave extra space for a minus sign in case of a negative number, but it seems to be leaving extra space in case of a positive number instead. Second, I'm curious about why the formatting is being done with leading zeros. I would have expected to see something like `{:>count$}` to left-pad with spaces.